crates/transpiler/src/passes/discretize_rotations.rs

```diff
@@ -0,0 +1,361 @@
+// This code is part of Qiskit.
+//
+// (C) Copyright IBM 2025
+//
+// This code is licensed under the Apache License, Version 2.0. You may
+// obtain a copy of this license in the LICENSE.txt file in the root directory
+// of this source tree or at http://www.apache.org/licenses/LICENSE-2.0.
+//
+// Any modifications or derivative works of this code must retain this
+// copyright notice, and modified files need to carry a notice indicating
+// that they have been altered from the originals.
+
+use pyo3::prelude::*;
+use pyo3::wrap_pyfunction;
+use std::f64::consts::PI;
+
+use qiskit_circuit::dag_circuit::DAGCircuit;
+use qiskit_circuit::operations::{OperationRef, Param, StandardGate};
+use rustworkx_core::petgraph::stable_graph::NodeIndex;
+
+static ROTATION_GATE_NAMES: [&str; 3] = ["rx", "ry", "rz"];
+
+const PI8: f64 = PI / 8.0;
```

Comment:
`std::f64::consts::FRAC_PI_8` is an already predefined constant you could use here

Reply:
done in [e7d61b1](https://github.com/Qiskit/qiskit/pull/15321/commits/e7d61b130d8d1c72433160bfde74786fb859b1fe)